thunder/torch/__init__.py

```diff
@@ -187,6 +187,68 @@ def is_cuda(a: TensorLike, /) -> bool:
 register_method("size", size)
 
 
+@torchsymbol(torch.Tensor.type, is_method=True)
+def type(a: TensorLike, dtype: None | str | dtypeLike = None, non_blocking: bool = False, /) -> str | TensorLike:
+    utils.check(
+        not non_blocking,
+        lambda: f"type(): `non_blocking==True` is currently not supported.",
+        exception_type=NotImplementedError,
+    )
+
+    DTYPE_STR = {
+        torch.float32: "torch.FloatTensor",
+        torch.float64: "torch.DoubleTensor",
+        torch.float16: "torch.HalfTensor",
+        torch.bfloat16: "torch.BFloat16Tensor",
+        torch.uint8: "torch.ByteTensor",
+        torch.int8: "torch.CharTensor",
+        torch.int16: "torch.ShortTensor",
+        torch.int32: "torch.IntTensor",
+        torch.long: "torch.LongTensor",
+        torch.bool: "torch.BoolTensor",
+    }
+
+    if dtype is None:
+        # returns the type of the input tensor in string
+        torch_dtype = to_torch_dtype(a.dtype)
+        torch_dtype = DTYPE_STR.get(torch_dtype)
+        if a.device.devicetype is devices.DeviceType.CUDA:
```

Comment:
Instead of splitting the string, map like this:

```
torch.bool: "BoolTensor"
```

and construct the final string like this: 

```
f"torch.{devicetype_str}{dtype_str}"
```

where `devicetype_str` can be acquired like this:

```
devicetype_str: str = ""
if a.device.devicetype is not devices.DeviceType.CPU:
  devicetype_str = f"{devices.devicetype_string(a.device.devicetype)}."
```

Reply:
Wow! You are amazing.